### PR TITLE
docs: add missing prop tables and prop descriptions (FE-5333)

### PR DESCRIPTION
**Pull request by grabkowski**
### Proposed behaviour

Add missing prop tables.
Add missing prop descriptions.

### Current behaviour

Prop tables missing in AccordionGroup, ActionPopoverMenuButton, ButtonBar, Fieldset, Slide.
Missing prop descriptions in AccordionGroup, ActionPopoverMenuButton, VerticalDivider.

### Checklist

- [x] Commits follow our style guide
~~- [ ] Related issues linked in commit messages if required~~
~~- [ ] Screenshots are included in the PR if useful~~
~~- [ ] All themes are supported if required~~
~~- [ ] Unit tests added or updated if required~~
~~- [ ] Cypress automation tests added or updated if required~~
- [x] Storybook added or updated if required
~~- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required~~
~~- [ ] Typescript `d.ts` file added or updated if required~~
~~- [ ] Related docs have been updated if required~~

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: src/components/flat-table/flat-table-header/flat-table-header.component.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ FlatTableHeader.propTypes = {
   ...paddingPropTypes,
   /** Content alignment */
   align: PropTypes.oneOf(["center", "left", "right"]),
+  /** Header comntent */
```
Comment (Parsium) [inline src/components/flat-table/flat-table-header/flat-table-header.component.js:63]:
```suggestion
  /** Header component */
```

File: src/components/flat-table/flat-table-header/flat-table-header.d.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export interface FlatTableHeaderProps extends PaddingProps {
   align?: "left" | "center" | "right";
   /** If true sets alternative background color */
   alternativeBgColor?: boolean;
+  /** Header comntent */
```
Comment (Parsium) [inline src/components/flat-table/flat-table-header/flat-table-header.d.ts:11]:
```suggestion
  /** Header component */
```

File: src/components/carousel/carousel.stories.mdx [hunk 3]
```diff
@@ -166,3 +166,5 @@ import { Carousel, Slide } from "carbon-react/lib/components/carousel";
 ### Carousel
 
 <ArgsTable of={Carousel} />
+
+<ArgsTable of={Slide} />
```
Comment (Parsium) [inline src/components/carousel/carousel.stories.mdx:170]:
```suggestion

### Slide
<ArgsTable of={Slide} />
```

Might be worth adding a header here so its clear who's props these are

File: src/components/flat-table/flat-table-header/flat-table-header.d.ts (hunk 2 above)
Comment (grabkowski) [reply src/components/flat-table/flat-table-header/flat-table-header.d.ts:11]: should be `Header content` :)

File: src/components/flat-table/flat-table-header/flat-table-header.component.js (hunk 1 above)
Comment (grabkowski) [reply src/components/flat-table/flat-table-header/flat-table-header.component.js]: `Header content`

**Review by Parsium — APPROVED** [review]

**Review by divyajindel — APPROVED** [review]

**Pull request merged**